tests/tests/eth_provider.rs

```diff
@@ -809,6 +812,179 @@ async fn test_send_raw_transaction_wrong_signature(#[future] katana: Katana, _se
+        ..Default::default()
+    };
+
+    // Initialize state override with the EOA address having a high balance
+    let mut state_override1: HashMap<Address, AccountOverride> = HashMap::new();
+    state_override1
+        .insert(eoa_address, AccountOverride { balance: Some(U256::from(1_000_000_000)), ..Default::default() });
+
+    // Perform the first call with state override and high balance
+    // The transaction should succeed
+    let _ = eth_provider
+        .call(request1, None, Some(state_override1.clone()), None)
+        .await
+        .expect("Failed to call for a simple transfer");
+
+    // Create the second transaction request with a higher value
+    let request2 = TransactionRequest {
+        from: Some(eoa_address),
+        to: Some(TxKind::Call(Address::ZERO)),
+        gas: Some(21000),
+        gas_price: Some(10),
+        value: Some(U256::from(1_000_000_001)),
+        ..Default::default()
+    };
+
+    // Attempt to call and handle the result
+    match eth_provider.call(request2, None, Some(state_override1.clone()), None).await {
+        // If call succeeds, panic as an error was expected
+        Ok(_) => panic!("Expected an error but got a success response"),
+        // If call fails, check if the error is due to insufficient funds
+        Err(e) => match e {
+            EthApiError::Transaction(TransactionError::Overrides(err)) => {
+                assert_eq!(
+                    err.to_string(),
+                    "transaction validation error: lack of funds (1000000000) for max fee (1000210001)",
+                );
+            }
+            // If a different error occurs, panic
+            _ => panic!("Expected a transaction error but got a different error: {e:?}"),
+        },
+    }
+
+    // Define a Solidity contract interface for the counter
+    sol! {
+        #[sol(rpc)]
+        contract CounterContract {
+            function incForLoop(uint256 iterations) external;
+        }
+    }
+
+    // Prepare the calldata for invoking the incForLoop function with 5 iterations
+    let calldata = CounterContract::incForLoopCall { iterations: U256::from(5) }.abi_encode();
+
+    // Define the transaction request for invoking the incForLoop function
+    let request3 = TransactionRequest {
+        from: Some(eoa_address),
+        to: Some(TxKind::Call(contract_address)),
+        gas: Some(210_000),
+        gas_price: Some(1_000_000_000_000_000_000_000),
+        value: Some(U256::ZERO),
+        nonce: Some(2),
+        input: TransactionInput { input: Some(calldata.clone().into()), data: None },
+        ..Default::default()
+    };
+
+    // Perform the call to invoke the incForLoop function without state override
+    let _ = eth_provider
+        .call(request3, None, None, None)
+        .await
+        .expect("Failed to call incForLoopCall in Kakarot constract");
+
+    // Define another Solidity contract with a different interface and bytecode
+    sol! {
```

Comment:
this test is very nice, I would basically:
- generate a random address and fund it via state override.
- generate the random bytecode as you did.
- call the contract using the random generated address.

TLDR: only keep this lower test as the override test and move the previous test to an `eth_call` test

Reply:
Yes I agree with you. I wanted to do a single test to avoid restarting Katana several times but ultimately it is too big and these tests are too important to neglect.

So I separated into several tests:
- One without any state override just to check that a simple transfer passes
- One with state override on transfer with success because the random account has a balance **greater** than what must be transferred.
- One with state override on transfer with failure because the random account has a balance **lower** than what must be transferred.
- One with modification of the bytecode after a first successful call to be sure that the state override overwrites the settings
- One with modification of the bytecode and call to the new bytecode to be sure that everything works well on a call after state overrides.


Normally, block overrides are also automatically included but not a priority so we do not test them on this PR which is already quite big. I suggest opening another issue solely dedicated to testing block overrides.